alignment points

NSoiffer opened the pull request:
Big changes to maligngroup/malignmark to eliminate their attributes and simplify the description.

Also, malignmark can no longer occur at any point, including inside tokens.

NSoiffer committed:
starting work on alignment rewrite

NSoiffer committed:
more changes to malgingroup/malignmark

NSoiffer committed:
Big changes to maligngroup/malignmark to eliminate their attributs and simplify the description.
Also, malignmark can no longer occur at any point, including inside tokens.

davidcarlisle commented on src/presentation-markup.html:
```diff
@@ -7387,6 +7384,23 @@ <h5 id="presm_mtdatts">Attributes</h5>
+      With one significant exception, <code class="emptytag">&lt;maligngroup/&gt;</code> and <code class="emptytag">&lt;malignmark/&gt;</code>
+      have had minimal adoption and implementation.
+      The one exception only uses the basics of alignment. Because of this, alignment in MathML is significantly
+      simplified to align with the current usage and make future implemntation simplier. In particular, the following simplifications are made:
```
implementation simpler

NSoiffer replied:
fixed

davidcarlisle commented on src/presentation-markup.html:
```diff
@@ -7482,61 +7483,49 @@ <h5>Description</h5>
-    later. In the example above, each equation would have one
+    <h5 id="presm_specify_align_group">Specifying alignment groups</h5>
+    <p>Each part of expression to be aligned should be in an <code class="element">maligngroup</code>.
+       The point of alignment is the left edge (right edge for RTL) of the element that follows an <code class="element">maligngroup</code> element
```
(right edge if `<code>dir="rtl"</code>`)

NSoiffer replied:
fixed

davidcarlisle commented on src/presentation-markup.html:
```diff
@@ -7578,55 +7567,40 @@ <h5>Specifying alignment groups</h5>
-    element in an element whose arguments have positional significance, it
-    is necessary to introduce a new <code class="element">mrow</code> element
-    containing just the <code class="element">maligngroup</code> element and the child element
-    it precedes in order to preserve the proper expression structure.  For
```
we could make `<mfrac><mn>1</mn><malignmark/></mfrac>` schema invalid,  previously they were valid as malignmark and maligngroup listed in PresentationExpression so allowed anywhere but they could be treated like `<none/>` and only allowed in specific contexts. The mathml4-legacy schema can make them valid again (and add attributes back, etc...

NSoiffer replied:
Sounds reasonable.

I won't mark this as resolved since you have some work to do.

davidcarlisle commented on src/presentation-markup.html:
```diff
@@ -7691,382 +7665,29 @@ <h5>Table cells that are not divided into alignment groups</h5>
     this situation, but it is not an error, and should trigger no warnings
     by default. <span>The rationale for this is that</span> it would
     be inconvenient to have to remove all
     unnecessary <code class="element">malignmark</code> elements from
```
Why not make it an error?

NSoiffer replied:
It is more consistent with the "malignmark/maligngroup" can occur anywhere philosophy (ignore if not appropriate) to ignore them.

davidcarlisle replied:
OK I suppose (personally I'd make them invalid, making it more likely editors avoid this, browsers silently ignore invalid markup in any case) But not a battle I want to fight so I'll go with the flow here, marking resolved

davidcarlisle reviewed (commented):
I suspect we should remove the features rather than deprecate.

Currently there is a broken link as `groupalign` is described linking to `#group-alignment-list-list` which no longer exists, but more generally I think "deprecated" means "if you use it, it should work, but the feature may be removed in future" So the benefit of simpler schema and simpler text not gained until MathML5.

So "deprecation" would mean we still have to describe the old full version, but then advise it isn't used.

I think we should treat this as if it had been deprecated in MathML3,  so that `groupalign=` or `<mn><maligmark/></mn>` can be marked invalid in mathml4 unless using `mathml4-legacy` schema.

davidcarlisle commented:
@NSoiffer 's branch is visible at

https://davidcarlisle.github.io/mathml/#deprecation-notice

NSoiffer committed:
Address @davidcarlisle's comments.

Beside some bug fixes, I've changed "deprecated" to "removed" and deleted the appropriate attrs.

It is likely some external references need to be updated.

davidcarlisle reviewed (approved):
schema is (still) in mathml-refresh github so I think all done here, I'll merge. I hope to update schema before the call this (UK) evening
